Beta_Integrations/GmailSingleUser/GmailSingleUser.py

```diff
@@ -0,0 +1,947 @@
+
+    result = client.send_mail(emailto, EMAIL, subject, body, entry_ids, cc, bcc, htmlBody,
+                              replyTo, file_names, attchCID, transientFile, transientFileContent,
+                              transientFileCID, additional_headers, template_param)
+    return client.sent_mail_to_entry('Email sent:', [result], emailto, EMAIL, cc, bcc, htmlBody, body, subject)
+
+
+'''FETCH INCIDENTS'''
+
+
+def fetch_incidents(client):
+    user_key = 'me'
+    query = '' if params['query'] is None else params['query']
+    last_run = demisto.getLastRun()
+    last_fetch = last_run.get('gmt_time')
+    # handle first time fetch - gets current GMT time -1 day
+    if last_fetch is None:
+        last_fetch, _ = parse_date_range(date_range=FETCH_TIME, utc=True, to_timestamp=False)
+        last_fetch = str(last_fetch.isoformat()).split('.')[0] + 'Z'
+
+    last_fetch = datetime.strptime(last_fetch, '%Y-%m-%dT%H:%M:%SZ')
+    current_fetch = last_fetch
+    service = client.get_service('gmail', 'v1')
+
+    query += last_fetch.strftime(' after:%Y/%m/%d')
+    LOG('GMAIL: fetch parameters:\nuser: %s\nquery=%s\nfetch time: %s' %
+        (user_key, query, last_fetch, ))
+
+    result = service.users().messages().list(
+        userId=user_key, maxResults=100, q=query).execute()
+
+    incidents = []
+    # so far, so good
+    LOG('GMAIL: possible new incidents are %s' % (result, ))
+    for msg in result.get('messages', []):
+        msg_result = service.users().messages().get(
+            id=msg['id'], userId=user_key).execute()
+        incident = client.mail_to_incident(msg_result, service, user_key)
+        temp_date = datetime.strptime(
+            incident['occurred'], '%Y-%m-%dT%H:%M:%SZ')
+        # update last run
+        if temp_date > last_fetch:
+            last_fetch = temp_date + timedelta(seconds=1)
+
+        # avoid duplication due to weak time query
+        if temp_date > current_fetch:
+            incidents.append(incident)
+
+    demisto.info('extract {} incidents'.format(len(incidents)))
+    demisto.setLastRun({'gmt_time': last_fetch.isoformat().split('.')[0] + 'Z'})
+    return incidents
+
+
+def auth_link_command(client: Client):
+    link, challange = client.generate_auth_link()
+    markdown = f"""
+## Gmail Auth Link
+Please follow the following **[link]({link})**.
+
+After Completing the authentication process, copy the received code
+to the **Auth Code** configuration parameter of the integration instance.
+Save the integration instance and then run *!gmail-auth-test* to test that
+the authentication is properly set.
+    """
+    return {
+        'ContentsFormat': formats['text'],
+        'Type': entryTypes['note'],
+        'Contents': markdown,
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': markdown,
+    }
+
+
+def auth_test_command(client):
+    client.search('me', '', '', '', '', '', '', '', '',
+                  '', [], 10, '', False, False)
```

Comment:
most of the values here are the default ones, why pass them?

Reply:
Let's just leave this for now.